services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/bolts/LinkOperationsBolt.java

```diff
@@ -53,14 +68,119 @@ public LinkOperationsBolt(PersistenceManager persistenceManager) {
     }
 
     private List<IslInfoData> getAllLinks() {
-        return repositoryFactory.createIslRepository().findAll().stream()
+        IslRepository islRepository = repositoryFactory.createIslRepository();
+        return islRepository.findAll().stream()
                 .map(IslMapper.INSTANCE::map)
                 .collect(Collectors.toList());
     }
 
     private List<LinkPropsData> getLinkProps(LinkPropsGet request) {
-        // TODO: should be re-implemented in the scope of TE refactoring.
-        return Collections.emptyList();
+        LinkPropsRepository linkPropsRepository = repositoryFactory.createLinkPropsRepository();
+
+        Integer srcPort = request.getSource().getPortNumber();
+        SwitchId srcSwitch = request.getSource().getDatapath();
+        Integer dstPort = request.getDestination().getPortNumber();
+        SwitchId dstSwitch = request.getDestination().getDatapath();
+
+        Collection<LinkProps> linkProps = linkPropsRepository.findByEndpoints(srcSwitch, srcPort, dstSwitch, dstPort);
+        return linkProps.stream()
+                .map(LinkPropsMapper.INSTANCE::map)
+                .map(LinkPropsData::new)
+                .collect(Collectors.toList());
+    }
+
+    private LinkPropsResponse putLinkProps(LinkPropsPut request) {
+        LinkPropsRepository linkPropsRepository = repositoryFactory.createLinkPropsRepository();
+        IslRepository islRepository = repositoryFactory.createIslRepository();
+
+        transactionManager.begin();
+        try {
+            LinkProps linkPropsToSet = LinkPropsMapper.INSTANCE.map(request.getLinkProps());
+
+            Collection<LinkProps> existingLinkProps = linkPropsRepository.findByEndpoints(
+                    linkPropsToSet.getSrcSwitchId(), linkPropsToSet.getSrcPort(),
+                    linkPropsToSet.getDstSwitchId(), linkPropsToSet.getDstPort());
+
+            LinkProps linkProps;
+            if (!existingLinkProps.isEmpty()) {
+                linkProps = existingLinkProps.iterator().next();
+                if (linkPropsToSet.getCost() != null) {
+                    linkProps.setCost(linkPropsToSet.getCost());
+                }
+                if (linkPropsToSet.getMaxBandwidth() != null) {
+                    linkProps.setMaxBandwidth(linkPropsToSet.getMaxBandwidth());
+                }
+                linkProps.setTimeModify(Instant.now());
+            } else {
+                linkProps = linkPropsToSet;
+                Instant timestamp = Instant.now();
+                linkProps.setTimeCreate(timestamp);
```

Comment:
You do not check that all required fields are set (srcSwitch, srcPort, dstPort, dstSwitch).